packages/core/lib/utils.js

```diff
@@ -75,7 +75,7 @@ exports.placeholdify = (config) => {
 exports.environmentalize = (_config) => {
   const _env = {};
   const env = global._env || process.env;
-  const regExp = /\[(\w+?)\]/g;
+  const regExp = /^\[([^=]+)=?([^\]]+)?\]$/g;
```

Comment:
With the start/end anchors we restrict where an env var can be found.
Previously patterns such as this were possible: `"apiUrl": "[SERVER_URL]/api"`.
I would propose to change the regex to something like this: `\[([a-zA-Z_][a-zA-Z0-9_]*)(?:=(.+))?\]`.
I did some quick tests and it seems to work okay-ish - if that is not enough for us, we need to implement a small parse function that holds state and reads each character one by one in a loop...

Reply:
Check out: https://regexr.com/4jvpr